lib/phoenix/test/channel_test.ex

```diff
@@ -127,8 +127,33 @@ defmodule Phoenix.ChannelTest do
 
   alias Phoenix.Socket
   alias Phoenix.Socket.Message
+  alias Phoenix.Socket.Broadcast
+  alias Phoenix.Socket.Reply
   alias Phoenix.Channel.Server
 
+  defmodule NoopSerializer do
+    def encode!(%Reply{} = reply) do
+      %Message{
+        topic: reply.topic,
+        event: "phx_reply",
+        ref: reply.ref,
+        payload: %{status: reply.status, response: reply.payload}
+      }
+    end
+    def encode!(%Broadcast{} = msg) do
+      %Message{
+        topic: msg.topic,
+        event: msg.event,
+        payload: msg.payload
+      }
+    end
+    def encode!(%Message{} = msg) do
+      msg
+    end
+
+    def decode!(message, :text), do: message
```

Comment:
Should we pass still `:text` into `decode!`. It seems something the transport and serializer itself should figure out? Can we be possibly leaking websockets concerns into all serializers?